cli/src/pcluster/config/cluster_config.py

```diff
@@ -1508,18 +1639,106 @@ def __init__(self, cloud_formation: ByosCloudFormationInfrastructure = None, **k
         self.cloud_formation = cloud_formation
 
 
+class ByosClusterSharedArtifact(Resource):
+    """Represent the ClusterSharedArtifact config for a BYOS plugin."""
+
+    def __init__(self, source: str, **kwargs):
+        super().__init__(**kwargs)
+        self.source = source
+
+    def _register_validators(self):
+        self._register_validator(UrlValidator, url=self.source, fail_on_https_error=True)
```

Comment:
we probably only want to warn in case we can't access the url of cluster shared artifacts(both if https or s3). This because maybe the CLI does not have access to such artifacts but cluster nodes will.
The warning message can be something like: "Failed to validate custom scheduler artifact '{url}'. Please make sure the URL is correct and cluster nodes have access to it."

Reply:
We have these places using UrlValidator 
```
cluster_template in dev_settings
chef_cookbook in dev_settings
node_package in dev_settings
aws_batch_cli_package in dev_settings
Script in CustomAction
template in ByosCloudFormationInfrastructure
Source in ByosClusterSharedArtifact
```
so only Source in ByosClusterSharedArtifact will give warning if CLI doesn't have access to it for s3 and https and others will give error?

Reply:
in theory all S3 urls that need to be accessible on the cluster nodes only can return a warning rather than an error, assuming that the cluster nodes have a policy to access those while the CLI does not.